pkg/app/piped/trigger/determiner.go

```diff
@@ -204,22 +204,43 @@ func (d *OnCommitDeterminer) ShouldTrigger(ctx context.Context, app *model.Appli
 	return true, nil
 }
 
-func isTouchedByChangedFiles(appDir string, changes []string, changedFiles []string) (bool, error) {
+// isTouchedByChangedFiles checks whether this application changed files can trigger a new deployment or not (considered as "touched")
+// The logic of watching files pattern contains both "includes" and "excludes" filter and be implemented as flow:
+//  1. If both includes & excludes are empty, app is considered as touched
+//  2. If any of changed files are listed in excludes, app is NOT considered as touched
+//  3. If pass (2) and any of changed files are listed in includes, app is considered as touched
+func isTouchedByChangedFiles(appDir string, includes, excludes []string, changedFiles []string) (bool, error) {
 	if !strings.HasSuffix(appDir, "/") {
 		appDir += "/"
 	}
 
-	// If any files inside the application directory was changed
-	// this application is considered as touched.
-	for _, cf := range changedFiles {
-		if ok := strings.HasPrefix(cf, appDir); ok {
-			return true, nil
+	// In case includes and excludes do not contains any thing
+	// consider any files changed inside the application directory as touched.
```

Comment:
```suggestion
	// In case includes and excludes do not contain anything,
	// it's considered any files changed inside the application directory as touched.
```